addons/mass_mailing/static/src/js/unsubscribe.js

```diff
@@ -12,6 +12,8 @@ odoo.define('mass_mailing.unsubscribe', function (require) {
     var mailing_id = parseInt($("input[name='mailing_id']").val());
     var res_id = parseInt($("input[name='res_id']").val());
     var token = (location.search.split('token' + '=')[1] || '').split('&')[0];
+    var demo_mode = $('input[name="demo_mode"]').val();
```

Comment:
Mmmh ? I don't think we should support a "demo mode" ... unsubscribe from list is not a link we want people to browse.